packages/central-server/__tests__/sync/CentralSyncManager.test.js

```diff
@@ -1985,44 +1988,77 @@ describe('CentralSyncManager', () => {
       jest.resetModules();
     });
 
-    it('inserts incoming changes into snapshots', async () => {
-      const patient1 = await models.Patient.create(fake(models.Patient));
-      const patient2 = await models.Patient.create(fake(models.Patient));
-      const changes = [patient1, patient2].map((r) => ({
-        direction: SYNC_SESSION_DIRECTION.OUTGOING,
-        isDeleted: !!r.deletedAt,
-        recordType: 'patients',
-        recordId: r.id,
-        data: r.dataValues,
-      }));
+    it('does not record audit changelogs when isMobile is false during completePush', async () => {
+      await models.LocalSystemFact.set(FACT_CURRENT_SYNC_TICK, '16');
+      await models.Setting.set('audit.changes.enabled', true);
+      const facility = await models.Facility.create(fake(models.Facility));
+      const patient = await models.Patient.create(fake(models.Patient));
+      const program = await models.Program.create(fake(models.Program));
+      const clinician = await models.User.create(fakeUser());
 
-      jest.doMock('@tamanu/database/sync', () => ({
-        ...jest.requireActual('@tamanu/database/sync'),
-        insertSnapshotRecords: jest.fn(),
-      }));
+      const programRegistry = await models.ProgramRegistry.create({
+        ...fake(models.ProgramRegistry),
+        programId: program.id,
+      });
 
-      const centralSyncManager = initializeCentralSyncManager();
+      const patientProgramRegistrationData = {
+        ...fake(models.PatientProgramRegistration),
+        programRegistryId: programRegistry.id,
+        clinicianId: clinician.id,
+        patientId: patient.id,
+        facilityId: facility.id,
+      };
+      const changes = [
+        {
+          direction: SYNC_SESSION_DIRECTION.OUTGOING,
+          isDeleted: false,
+          recordType: 'patient_program_registrations',
+          recordId: crypto.randomUUID(),
+          data: patientProgramRegistrationData,
+        },
+      ];
 
-      const { insertSnapshotRecords } = require('@tamanu/database/sync');
+      const centralSyncManager = initializeCentralSyncManager({
+        sync: {
+          lookupTable: {
+            enabled: true,
+          },
+          maxRecordsPerSnapshotChunk: DEFAULT_MAX_RECORDS_PER_SNAPSHOT_CHUNKS,
+        },
+      });
       const { sessionId } = await centralSyncManager.startSession();
       await waitForSession(centralSyncManager, sessionId);
 
+      await centralSyncManager.setupSnapshotForPull(
+        sessionId,
+        {
+          since: 1,
+          facilityIds: [facility.id],
+        },
+        () => true,
+      );
```

Comment:
Why do we need to set up a pull? Is there something about the push that is dependent on it?